### PR TITLE
Replace master branch reference with main

### PR DESCRIPTION
**Pull request by kashifest**
Metal3 docs default branch is switched to `main`. Also most other repos have already switched their default branch to `main`. This PR replaces appropriate mentions of `master` branch to `main`.

### CODE REVIEW

**Review by fmuyassarov — APPROVED** [review]

File: design/baremetal-operator/bmo-part-of-capm3.md [hunk 1]
```diff
@@ -206,9 +206,9 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 namespace: capm3-system
 resources:
-- github.com/metal3-io/baremetal-operator/deploy/operator/?ref=master
-- github.com/metal3-io/baremetal-operator/deploy/crds/?ref=master
-- github.com/metal3-io/baremetal-operator/deploy/rbac/?ref=master
+- github.com/metal3-io/baremetal-operator/deploy/operator/?ref=main
+- github.com/metal3-io/baremetal-operator/deploy/crds/?ref=main
+- github.com/metal3-io/baremetal-operator/deploy/rbac/?ref=main
```
[the remark this answers is not in the document]
Comment (kashifest) [reply design/baremetal-operator/bmo-part-of-capm3.md:211]: Indeed. Which is why I was a bit hesitant to touch them. The doc that you are talking about is bmo-part-of-capm3 design doc which itself doesn't exist anymore. So lets keep it as it is.

**Pull request merged**